### ISSUE TITLE
Add minor formatting of view hashmap

### ISSUE THREAD

**Issue by flibbles** [felixhayashi/TW5-TiddlyMap#223]
This is a minor suggestion for a minor improvement.

Each view has a file along the lines `$:/plugins/felixhayashi/tiddlymap/graph/view/*/map.tid`where it stores a sizeable hashmap containing each node GUID and its corresponding location. That's cool. It works fine too. However, I've noticed that if one uses Tiddlywiki through NodeJS and they use versioning software (like me), then that file is kind of unwieldy. This is because JSON.stringify puts all the JSON on a single line by default. That's great for saving space, but it means that even the slightest change to the view causes git or svn to see that entire line as changed. This results in guaranteed conflicts, and also gigantic walls of JSON when looking through diffs.

If that text were formatted _just a little bit_, then this stops being a problem.

So in `lib.utils.js`, the `utils.writeFieldData` function changes to this:

```javascript
utils.writeFieldData = function(tiddler, field, data) {                         
  
  if(typeof data === "object") {                                                                                                
    var indentation = 0; 
    if (field == "text") {                                                      
      indentation = 1;                                                          
    }
    utils.setField(tiddler, field, JSON.stringify(data, null, indentation));    
  }                                                                             
                                                                                
};      
```

That's it.

## Pros

* Versioning conflicts are no longer a guaranteed thing in collaborative projects (for those strange, sad fools who work on Tiddlywiki collaboratively).
* What conflict _do_ come up are possible to resolve. Unlike resolving a single enormous line, which is a path to madness.
* git and svn aren't constantly rewriting the bulk of the file with every little change committed regarding the view.
* The view's map file is human readable (For the _particularly_ strange, sad fools who want to look at it, or even edit it). It also makes it more shell script friendly (I've done this once with awk, to expand the size of a view. I have a problem. I know.)

## Cons

* That file is about 20% larger. If I could, I'd use a setting that only introduced a single newline after each entry. That would be only 2% larger, but there's no easy or foolproof way of doing that.

That's it. Like I said, very tiny improvement.
-Flibbles

**Comment by felixhayashi** [comment]
Hi @flibbles 

thanks for laying this out. I'll add an option for formatting so the user can decide what he/she prefers.

-felix

**Comment by felixhayashi** [comment]
Hi @flibbles, I added a config param that you can use in the next release:

You will need to open the tiddler  `$:/plugins/felixhayashi/tiddlymap/config/sys/user`
and add:

    jsonIndentation: 1

-felix

**Comment by felixhayashi** [comment]
Hi @flibbles, this is now solved with v0.11.3 …
